napari/plugins/_npe2.py

```diff
@@ -364,20 +505,40 @@ def _npe2_manifest_to_actions(
                     subitem = _npe2_submenu_to_app_model(item)
                     submenus.append((menu_id, subitem))
 
-    actions: List[Action] = [
-        Action(
-            id=cmd.id,
-            title=cmd.title,
-            category=cmd.category,
-            tooltip=cmd.short_title or cmd.title,
-            icon=cmd.icon,
-            enablement=cmd.enablement,
-            callback=cmd.python_name or '',
-            menus=cmds.get(cmd.id),
-            keybindings=[],
-        )
-        for cmd in mf.contributions.commands or ()
-    ]
+    # Filter sample data commands as they are registered in
```

Comment:
Maybe here we should just add an if not already registered check? There's a lot of iterating and checking here when we could just skip if registered.

Reply:
Agree, this should be improved.
I am wondering if there could be any issue with simply checking if a command has already been registered with `app`. I say this partly because the only reason we picked up a bug in the code was because of an `ValueError: Command <> already registered` error. Potentially a typo in a plugin manifest, where the id is the same as one already registered, would be picked up by by the current code...?

Reply:
Ah also we register sample actions and all other actions at the same time (in `_register_manifest_actions`), so I don't think checking if it has already been registered would not work.

I have however amended so we only do on 'extra' iterating, to get all the sample data commands (since sample URI do not have a command). Open to suggestions.